### DISCUSSION TITLE
problem when english and persian characters are combined

### DISCUSSION THREAD

**Discussion by mojtaba-soltani** [QuestPDF/QuestPDF#744] [General]
hi everybody
i have a problem when there are english and persian characters in my text. consider the below text :
"مانیتور-sumsong-125-اینچ"
this is real text. but when i generate pdf the below text is shown in pdf :
"مانتیورینگ-521-gnosmus-اینچ"

as u can see the english characters and numbers are reversed. 

how can i solve my problem?
thanks

**Comment by girlpunk** [comment]
To mix text directions, characters in different directions must be sent in different spans, for example:

```csharp
container.Text(text =>
{
    text.Span("اینچ");
    text.Span("-125-sumsong-");
    text.Span("مانیتور");
});
```

**Reply by mojtaba-soltani** [reply]
thanks for answering me but the data is coming from the database. I have no control over data.
Sometimes all text is in English like 725-Lg sometimes in Persian kike "سامسونگ - 425"
and sometimes it is a combination of both English and Persian like "مانتیورینگ-521-gnosmus-اینچ" and sometimes there is no number in the text
thanks

**Reply by MarcinZiabek** [reply]
I am happy to share that extended multilingual support will be available as part of this high-priority effort very soon: https://github.com/QuestPDF/QuestPDF/discussions/622 😀

**Reply by girlpunk** [reply]
This has also been discussed in detail in #445 

I believe this comment still applies
> The best workaround I can come up with would be to split the input string based on the codepage of characters in it, as detailed [here](https://stackoverflow.com/a/45628450/1132448), and then render the result as separate spans with or without the direction override as needed.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
